server/src/main/java/org/elasticsearch/common/time/FormatNames.java

```diff
@@ -108,11 +112,19 @@
     STRICT_YEAR_MONTH_DAY("strictYearMonthDay", "strict_year_month_day");
 
     private static final Set<String> ALL_NAMES = Arrays.stream(values())
-                                                       .flatMap(n -> Stream.of(n.snakeCaseName, n.camelCaseName))
-                                                       .collect(Collectors.toSet());
+        .flatMap(n -> Stream.of(n.snakeCaseName, n.camelCaseName))
+        .collect(Collectors.toSet());
+
     private final String camelCaseName;
     private final String snakeCaseName;
 
+    // DateFormatters and FormatNames are being used even before the logging is initialized.
+    // If LogManager.getLogger is called before logging config is loaded
+    // it results in errors sent to status logger and startup to fail.
+    // Hence a lazy initialization.
+    private static final LazyInitializable<DeprecationLogger, RuntimeException> deprecationLogger
+        = new LazyInitializable(() -> new DeprecationLogger(LogManager.getLogger(FormatNames.class)));
```

Comment:
as per comment. Some classes - which are used before loading log4j config  - are using `DateFormat.forPattern` to get a date format. see `XContentElasticsearchExtension` 
This is fine, but if I try to create a logger, before logging config is loaded I will get an exception (no config, writing to status logger) this is later checked and startup will fail.
I am not sure we can move loading log4j config as early on startup. 
At the same time, this logging is a temporary. We will remove it in `8` (unless we want some more detailed exception)
@rjernst WDYT?